### ISSUE TITLE
Twice normalization of local distance

### ISSUE THREAD

**Issue by marvis** [huanghoujing/AlignedReID-Re-Production-Pytorch#2]
Hi @huanghoujing ,

I find that the distance between local features performs twice normalization. 
1: https://github.com/huanghoujing/AlignedReID-Re-Production-Pytorch/blob/master/aligned_reid/tri_loss/model/Loss.py#L169
2: https://github.com/huanghoujing/AlignedReID-Re-Production-Pytorch/blob/master/aligned_reid/tri_loss/model/Loss.py#L75

After twice normalize the final distance is less than 1. I wonder whether this is a problem?

Best,

**Comment by huanghoujing** [comment]
The first and second normalization are before and after computing euclidean distance, respectively. The second one is described in Equation (1) of the paper. The first one is added by myself -- I once found large euclidean distance caused saturation after Equation (1) and the gradient is tooooooo small to move the model forward. So I add L2 normalization before computing euclidean distance. However, distance of two unit-length vectors falls in range [0, 2], thus after Equation (1), the distance is in range [0, 0.76]. The necessity of the first normalization should be further verified by experiment.

**Comment by marvis** [comment]
@huanghoujing  Thanks for your reply and looking forward for your further results.

One more question.  When training on market1501 using resnet50, the results in open-reid is 85.1 for triplet and 81.4 for softmax.  But your baseline result (GL-NF) is 81.53%.  Do you use softmax training as the baseline? 

Best,

**Comment by marvis** [comment]
@huanghoujing Thanks for your reply and looking forward for your results.

**Comment by huanghoujing** [comment]
GL-NF only uses Triplet Loss with Global distance. The difference to open-reid triplet loss example includes 
- whether to use standard Market1501 train/test split, false in open-reid, but true here
- whether to use an extra embedding layer (128-dim) after average pooling (2048-dim), used in open-reid, but here not
- the epoch at which the learning rate starts to decay, epoch 100 for open-reid and 75 here

Thanks :)
